src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocketOptions.cs

```diff
@@ -143,6 +160,52 @@ public TimeSpan KeepAliveInterval
             }
         }
 
+        internal int ReceiveBufferSize => _receiveBufferSize;
+        internal int SendBufferSize => _sendBufferSize;
+        internal ArraySegment<byte>? Buffer => _buffer;
+
+        public void SetBuffer(int receiveBufferSize, int sendBufferSize)
+        {
+            ThrowIfReadOnly();
```

Comment:
There are already settings/members of ClientWebSocketOptions ignored by the various implementations, including the WinHTTP-based implementation.  Since buffers are invisible from a behavioral perspective, I don't see a problem for the foreseeable future with implementations simply ignoring the buffer sizes or even the provided buffer if it doesn't suit the implementation.  Over time the implementation could choose to take advantage of it if it made sense.  For the managed implementation, I updated it to respect the receive buffer size, as well as the buffer itself if it meets certain criteria (namely that the array segment is for the whole underlying array).